### PR TITLE
chore(wallet) remove legacy code to fetch details for displaying headers

### LINKED ISSUE #11598: Remove fetching of transaction and multi-transaction details for displaying headers - activity filtering optimization
Blocked by https://github.com/status-im/status-desktop/issues/11597

### PR DESCRIPTION
**Pull request by stefandunca**

### CODE REVIEW

**Comment by stefandunca** [comment]
Great find, fixed it. Please have another look.

File: src/app/modules/main/wallet_section/activity/controller.nim [hunk 1]
```diff
@@ -74,136 +74,48 @@ QtObject:
...
+                                self.currencyService.getCurrencyFormat(symbol))
     for backendEntry in backendEntities:
-      case backendEntry.payloadType:
+      var ae: entry.ActivityEntry
+      case backendEntry.getPayloadType():
         of MultiTransaction:
-          multiTransactionsIds.add(backendEntry.id)
+          let extraData = self.buildMultiTransactionExtraData(backendEntry)
+          ae = entry.newMultiTransactionActivityEntry(backendEntry, extraData, amountToCurrencyConvertor)
         of SimpleTransaction:
```
Comment (dlipicar) [inline src/app/modules/main/wallet_section/activity/controller.nim:99]: nit: this case and the one below look identical, could comma-separate the two enum values

File: src/app/modules/main/wallet_section/activity/entry.nim [hunk 2]
```diff
@@ -22,22 +20,10 @@ type
...
 
-# It is used to display an activity history entry in the QML UI
-#
-# TODO remove this legacy after the NFT is served async; see #11598
-#
-# Looking into going away from carying the whole detailed data and just keep the required data for the UI
-# and request the detailed data on demand
-#
-# Outdated: The ActivityEntry contains one of the following instances transaction, pending transaction or multi-transaction
+# User to display an activity history header entry in the QML UI
```
Comment (dlipicar) [inline src/app/modules/main/wallet_section/activity/entry.nim:23]: typo User->Used

File: src/app/modules/main/wallet_section/activity/entry_details.nim [hunk 3]
```diff
@@ -33,41 +38,54 @@ QtObject:
...
     if len(maxFeePerGas) > 0 and len(gasLimit) > 0:
       let maxTotalFees = getMaxTotalFees(maxFeePerGas, gasLimit)
       result.maxTotalFees = valueConvertor(stint.fromHex(UInt256, maxTotalFees), "Gwei")
-    else:
-      result.maxTotalFees = newCurrencyAmount()
 
     if e.hasKey(totalFeesField) and e[totalFeesField].kind != JNull:
       let totalFees = e[totalFeesField].getStr()
-      result.totalFees = valueConvertor(stint.fromHex(UInt256, totalFees), "Gwei")
+      let resTotalFees = valueConvertor(stint.fromHex(UInt256, totalFees), "Gwei")
```
Comment (dlipicar) [inline src/app/modules/main/wallet_section/activity/entry_details.nim:86]: we probably (should) have `Gwei` defined as a constant somewhere

**Review by dlipicar — APPROVED** [review]
Awesome work, so nice to see the final pieces come into place

**Pull request merged**